pkg/diags/diags.go

```diff
@@ -0,0 +1,26 @@
+package diags
+
+import (
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/api/errors"
+	"golang.org/x/net/context"
+)
+
+// Provider is collection of APIs that implement the SDK OpenStorageDiags service
+type Provider interface {
+	// Collect collects diagnostics based on the provided request
+	Collect(ctx context.Context, in *api.SdkDiagsCollectRequest) (*api.SdkDiagsCollectResponse, error)
```

Comment:
Instead of using a copy of the interface, you can use the interface created in `api.pb.gb` here. See `pkg/role` as an example.

Reply:
https://github.com/libopenstorage/openstorage/blob/0c2f335282d9ed2bb26b5ac9f5f2f2d744271274/pkg/role/role.go#L27

Reply:
Good idea. Done.